public/docs/ts/_cache/tutorial/toh-pt5.jade

```diff
@@ -148,172 +168,220 @@ code-example(language="bash").
+
 :marked
   The  *AppComponent* is now attached to a router and displaying routed views.
   For this reason and to distinguish it from other kinds of components,
   we call this type of component a *Router Component*.
 
-
 :marked
   ## Add a *Dashboard*
+
   Routing only makes sense when we have multiple views. We need another view.
 
   Create a placeholder `DashboardComponent` that gives us something to navigate to and from.
-+makeExample('toh-5/ts/app/dashboard.component.1.ts',null, 'app/dashboard.component.ts (v1)')(format=".")
+
++makeExcerpt('app/dashboard.component.1.ts (v1)', '')
+
 :marked
   We’ll come back and make it more useful later.
 
   ### Configure the dashboard route
-  Go back to `app.routing.ts` and teach it to navigate to the dashboard.
 
-  Import the `DashboardComponent` so we can reference it in the dashboard route definition.
+  Go back to `!{_appRoutingTsVsAppComp}` and teach it to navigate to the dashboard.
 
-  Add the following `'Dashboard'` route definition to the `Routes` array of definitions.
-+makeExample('toh-5/ts/app/app.routing.2.ts','dashboard-route', 'app/app.routing.ts (Dashboard route)')(format=".")
+  Import `DashboardComponent` and 
+  add the following route definition to the `!{_RoutesVsAtRouteConfig}` !{_array} of definitions.
 
-:marked
-  We also need to add the `DashboardComponent` to our root NgModule's `declarations`.
+- var _file = _docsFor == 'dart' ? 'lib/app_component.dart' : 'app/app.routing.ts'
++makeExcerpt(_file + ' (Dashboard route)', 'dashboard')
 
-+makeExample('toh-5/ts/app/app.module.3.ts','dashboard-declaration', 'app/app.module.ts (Dashboard declaration)')(format=".")
-
-.l-sub-section
++ifDocsFor('ts|js')
   :marked
-    **Redirect**
+    Also import and add `DashboardComponent` to our root NgModule's `declarations`.
 
-    We want the app to show the dashboard when it starts and
-    we want to see a nice URL in the browser address bar that says `/dashboard`.
-    Remember that the browser launches with `/` in the address bar.
-    We can use a redirect route to make this happen.
+  +makeExcerpt('app/app.module.ts', 'dashboard')
 
-+makeExample('toh-5/ts/app/app.routing.2.ts','redirect-route', 'app/app.routing.ts (Redirect route)')(format=".")
+:marked
+  #### !{_redirectTo}
 
-.l-sub-section
+  We want the app to show the dashboard when it starts and
+  we want to see a nice URL in the browser address bar that says `/dashboard`.
+  Remember that the browser launches with `/` in the address bar.
+
+block redirect-vs-use-as-default
   :marked
-    Learn about the *redirects* in the [Routing](../guide/router.html#!#redirect) chapter.
+    We can use a redirect route to make this happen. Add the following
+    to our array of route definitions:
+
+  +makeExcerpt('app/app.routing.ts','redirect')
+
+  .l-sub-section
+    :marked
+      Learn about the *redirects* in the [Routing](../guide/router.html#!#redirect) chapter.
 
 :marked
```

Comment:
Add a heading above this, since you're no longer talking about useAsDefault/redirectTo. Perhaps:

#### Add navigation to the template